pkg/apis/batch/validation/validation.go

```diff
@@ -389,6 +399,92 @@ func validatePodFailurePolicyRuleOnExitCodes(onExitCode *batch.PodFailurePolicyO
+		}
+		var err error
+		if totalIndexes, err = validateIndexesFormat(*criteria.SucceededIndexes, int(*spec.Completions)); err != nil {
+			allErrs = append(allErrs, field.Invalid(succeededIndexes, *criteria.SucceededIndexes, fmt.Sprintf("error parsing succeededIndexes: %s", err.Error())))
+		}
+	}
+	if criteria.SucceededCount != nil {
+		succeededCountPath := criteriaPath.Child("succeededCount")
+		if *criteria.SucceededCount < 0 {
+			allErrs = append(allErrs, apivalidation.ValidateNonnegativeField(int64(*criteria.SucceededCount), succeededCountPath)...)
+		}
+		if *criteria.SucceededCount > *spec.Completions {
+			allErrs = append(allErrs, field.Invalid(succeededCountPath, *criteria.SucceededCount, fmt.Sprintf("must be less than or equal to %d (the number of specified completions)", *spec.Completions)))
+		}
+		if totalIndexes != nil && int(*criteria.SucceededCount) > *totalIndexes {
+			allErrs = append(allErrs, field.Invalid(succeededCountPath, *criteria.SucceededCount, fmt.Sprintf("must be less than or equal to %d (the number of indexes in the specified succeededIndexes field)", *totalIndexes)))
+		}
+	}
+	return allErrs
+}
+
+// Extended the [validateIndexesFormat](https://github.com/kubernetes/kubernetes/pull/123273)
+// TODO: Once #123273 is merged into the master, we should rebase these commits.
+func validateIndexesFormat(indexesStr string, completions int) (*int, error) {
+	if indexesStr == "" {
+		return nil, nil
+	}
+	var lastIndex *int
+	var total int
+	for _, intervalStr := range strings.Split(indexesStr, ",") {
+		limitsStr := strings.Split(intervalStr, "-")
+		if len(limitsStr) > 2 {
+			return nil, fmt.Errorf("the fragment %q violates the requirement that an index interval can have at most two parts separated by '-'", intervalStr)
+		}
+		x, err := strconv.Atoi(limitsStr[0])
+		if err != nil {
+			return nil, fmt.Errorf("cannot convert string to integer for index: %q", limitsStr[0])
+		}
+		if x >= completions {
+			return nil, fmt.Errorf("too large index: %q", limitsStr[0])
+		}
+		if lastIndex != nil && *lastIndex >= x {
+			return nil, fmt.Errorf("non-increasing order, previous: %d, current: %d", *lastIndex, x)
+		}
+		lastIndex = &x
+		total += 1
+		if len(limitsStr) > 1 {
+			y, err := strconv.Atoi(limitsStr[1])
+			if err != nil {
+				return nil, fmt.Errorf("cannot convert string to integer for index: %q", limitsStr[1])
+			}
+			if y >= completions {
+				return nil, fmt.Errorf("too large index: %q", limitsStr[1])
+			}
+			if *lastIndex >= y {
+				return nil, fmt.Errorf("non-increasing order, previous: %d, current: %d", *lastIndex, y)
+			}
+			total += y - *lastIndex
+			lastIndex = &y
+		}
+	}
+	return &total, nil
+}
+
 // validateJobStatus validates a JobStatus and returns an ErrorList with any errors.
 func validateJobStatus(status *batch.JobStatus, fldPath *field.Path) field.ErrorList {
```

Comment:
This PR has impacts on status.  Particularly the relationship of various status conditions with the new condition being added for SuccessCriteriaMet.  https://github.com/kubernetes/kubernetes/pull/123273 established a good pattern for how to do this and a similar technique should be employed here.

Reply:
I'll try to check that PR.

Reply:
@deads2k Sorry, I couldn't find what you are pointing out. Could you give me some pointers?
@mimowo Could you help me with what we should apply #123273 's approach to this PR?

Reply:
Oh, I found your commit just now: https://github.com/kubernetes/kubernetes/pull/123273/commits/9f6244d47930f07ca2f7046d2b002c1cec836e38

Reply:
Done.

Reply:
@deads2k In this case we are not tightening existing fields or conditions. So I don't see why we need to follow the ratcheting code.